Fixes permission for cache json file from 600 to 644

rbertol opened the pull request:
##### SUMMARY

Until ansible-core 2.12 the facts cache file created by this module, have permission set as 644 which allow the other users to read the cache, since ansible-core 2.13, we create the temporary file, but we do not set the permission after renaming the temporary file.  Adding the line to set the permission to allow other users/groups read this file.

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

Permission when setting the cache to store in jsonfile `ANSIBLE_CACHE_PLUGIN=jsonfile` and `ANSIBLE_CACHE_PLUGIN_CONNECTION="$HOME/facts_cache"` at ansible-core-2.13+
```
facts_cache_2.14:
total 28K
-rw------- 1 root root 25K Feb 27 14:38 localhost
```

Permission when setting the cache to store in jsonfile `ANSIBLE_CACHE_PLUGIN=jsonfile` and `ANSIBLE_CACHE_PLUGIN_CONNECTION="$HOME/facts_cache"` at ansible-core-2.12
```
facts_cache_2.12:
total 24K
-rw-r--r-- 1 root root 24K Feb 27 14:35 localhost
```

sivel commented:
FWIW this is fixing https://github.com/ansible/ansible/issues/82683

nitzmahone commented:
Yeah, this is a specific instance of a more general problem around tempfiles clobbering permissions/ACLs/SELinux contexts, but in the absence of a more general solution, this is a good change to restore some of the old behavior.

Can you add a changelog fragment?

RudneiBertolJr commented:
Hey @nitzmahone, 

This is my first PR, if you have some examples of changelog fragment I am glad to do that.

rbertol committed:
Fixes permission for cache json file from 600 to 644

Until ansible-core 2.12 the facts cache file created by this module, have permission set as 644 which allow the other users to read the cache, since ansible-core 2.13, we create the temporary file, but we do not set the permission after renaming the temporary file.  Adding the line to set the permission to allow other users/groups read this file.

Created 82683-ansible-fact_cache-permissions-changed-after-ansible-coreupdate.yml changelog file.

rbertol commented:
@nitzmahone changelog fragment file created.

bcoca commented:
We probably need to add a permissions option to the 'file cache plugins', which can default to 644, but should be settable by the user, as their context will dictate the security required around the files.
